rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/assembler/BulkResultTasksRepresentationModelAssemblerTest.java

```diff
@@ -0,0 +1,170 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import pro.taskana.common.api.BulkOperationResults;
+import pro.taskana.common.api.TaskanaEngine;
+import pro.taskana.common.api.exceptions.ErrorCode;
+import pro.taskana.common.api.exceptions.InvalidArgumentException;
+import pro.taskana.common.api.exceptions.TaskanaException;
+import pro.taskana.rest.test.TaskanaSpringBootTest;
+import pro.taskana.task.api.TaskService;
+import pro.taskana.task.api.models.BulkResultTasks;
+import pro.taskana.task.internal.models.BulkResultTasksImpl;
+import pro.taskana.task.rest.models.BulkResultTasksRepresentationModel;
+import pro.taskana.workbasket.api.exceptions.NotAuthorizedOnWorkbasketException;
+import pro.taskana.workbasket.api.exceptions.WorkbasketNotFoundException;
+
+@TaskanaSpringBootTest
+class BulkResultTasksRepresentationModelAssemblerTest {
+
+  TaskanaEngine taskanaEngine;
+  TaskService taskService;
+  BulkResultTasksRepresentationModelAssembler assembler;
+
+  @Autowired
+  BulkResultTasksRepresentationModelAssemblerTest(
+      TaskanaEngine taskanaEngine,
+      TaskService taskService,
+      BulkResultTasksRepresentationModelAssembler assembler) {
+    this.taskanaEngine = taskanaEngine;
+    this.taskService = taskService;
+    this.assembler = assembler;
+  }
+
+  @Test
+  void should_ReturnEntity_When_ConvertingRepresentationModelToEntity() throws Exception {
+    // given
+    List<String> taskIds =
+        List.of(
+            "TKI:000000000000000000000000000000000003",
+            "TKI:000000000000000000000000000000000004",
+            "TKI:000000000000000000000000000000000039",
+            "TKI:000000000000000000000000000000000040");
+    BulkOperationResults<String, TaskanaException> result =
+        taskanaEngine.runAsAdmin(
+            () -> {
+              try {
+                return taskService.transferTasksWithOwner(
+                    "WBI:100000000000000000000000000000000006", taskIds, "user-1-1", true);
+              } catch (WorkbasketNotFoundException e) {
+                throw new RuntimeException(e);
+              } catch (NotAuthorizedOnWorkbasketException e) {
+                throw new RuntimeException(e);
+              }
+            });
+    Map<String, ErrorCode> newErrorMap = new HashMap<>();
+    for (Map.Entry<String, TaskanaException> entry : result.getErrorMap().entrySet()) {
+      newErrorMap.put(entry.getKey(), entry.getValue().getErrorCode());
+    }
+    BulkResultTasksRepresentationModel repModel = new BulkResultTasksRepresentationModel();
+    repModel.setSuccessfullyTransferredTaskIds(
+        List.of(
+            "TKI:000000000000000000000000000000000003",
+            "TKI:000000000000000000000000000000000004"));
+    repModel.setFailedTasksIdsErrorCode(newErrorMap);
+
+    // when
```

Comment:
Please remove //given //when //then from this test class

Reply:
I have removed them